core/math/vector3.h

```diff
@@ -400,14 +401,23 @@ void Vector3::zero() {
 	x = y = z = 0;
 }
 
-Vector3 Vector3::slide(const Vector3 &p_vec) const {
-
-	return p_vec - *this * this->dot(p_vec);
+// slide returns the component of the vector along the given plane, specified by its normal vector.
+Vector3 Vector3::slide(const Vector3 &p_n) const {
+#ifdef DEBUG_ENABLED
+	ERR_FAIL_COND_V(p_n.is_normalized() == false, Vector3());
+#endif
+	return *this - p_n * this->dot(p_n);
 }
 
-Vector3 Vector3::reflect(const Vector3 &p_vec) const {
+Vector3 Vector3::bounce(const Vector3 &p_n) const {
+	return -reflect(p_n);
+}
 
-	return p_vec - *this * this->dot(p_vec) * 2.0;
+Vector3 Vector3::reflect(const Vector3 &p_n) const {
+#ifdef
```

Comment:
Yeah, been a little busy during the weekend, sorry! Will push the update in a moment